### PR TITLE
[Meshmoving] porting to Amatrix

### PR DESCRIPTION
**Pull request by philbucher**
Porting the MeshMoving-App to Amatrix

Tests are running in Release and FullDebug

I am not sure abt the changes, please have a close look

fyi @AndreasWinterstein

### CODE REVIEW

File: applications/MeshMovingApplication/custom_elements/laplacian_meshmoving_element.cpp [hunk 1]
```diff
@@ -131,7 +131,7 @@ void LaplacianMeshMovingElement::CalculateLocalSystem(
     // Compute RHS
     CalculateDeltaPosition(delta_displacement, rCurrentProcessInfo);
     noalias(rRightHandSideVector) =
-        -prod(rLeftHandSideMatrix, delta_displacement);
+        (-1)*prod(rLeftHandSideMatrix, delta_displacement);
```
Comment (philbucher) [inline applications/MeshMovingApplication/custom_elements/laplacian_meshmoving_element.cpp]: @pooyan-dadvand I didn't see this elsewhere
Is there a better way?

Comment (AndreasWinterstein) [reply applications/MeshMovingApplication/custom_elements/laplacian_meshmoving_element.cpp]: Why do you want to change this to (-1) *...?

Comment (AndreasWinterstein) [reply applications/MeshMovingApplication/custom_elements/laplacian_meshmoving_element.cpp]: Why do you want to change this to (-1) *...?

Comment (philbucher) [reply applications/MeshMovingApplication/custom_elements/laplacian_meshmoving_element.cpp]: The current version does not compile with amatrix
Would be a nice feature though I think

Comment (pooyan-dadvand) [reply applications/MeshMovingApplication/custom_elements/laplacian_meshmoving_element.cpp]: Does `prod(-rLeftHandSideMatrix, delta_displacement);` works? (Note the '-' in front of LHS)

and would you please give me the error with minus in front of the prod?

Comment (philbucher) [reply applications/MeshMovingApplication/custom_elements/laplacian_meshmoving_element.cpp]: Thanks for your suggestion @pooyan-dadvand , now it works!

Here is still the error with minus in front of prod:
~~~
/home/philipp/software/Kratos_MasterBranch/applications/MeshMovingApplication/custom_elements/structural_meshmoving_element.cpp: In member function ‘virtual void Kratos::StructuralMeshMovingElement::CalculateLocalSystem(Kratos::StructuralMeshMovingElement::MatrixType&, Kratos::StructuralMeshMovingElement::VectorType&, Kratos::ProcessInfo&)’:
/home/philipp/software/Kratos_MasterBranch/applications/MeshMovingApplication/custom_elements/structural_meshmoving_element.cpp:274:37: error: no match for ‘operator-’ (operand type is ‘AMatrix::MatrixProductExpression<Kratos::Internals::Matrix<double, 0, 0>, Kratos::Internals::Matrix<double, 0, 1> >’)
     noalias(rRightHandSideVector) = -prod(rLeftHandSideMatrix, last_values);
                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/philipp/software/Kratos_MasterBranch/external_libraries/a_matrix/include/dense_storage.h:3:0,
                 from /home/philipp/software/Kratos_MasterBranch/external_libraries/a_matrix/include/matrix_storage.h:3,
                 from /home/philipp/software/Kratos_MasterBranch/external_libraries/a_matrix/include/matrix.h:8,
                 from /home/philipp/software/Kratos_MasterBranch/external_libraries/a_matrix/include/amatrix.h:1,
                 from /home/philipp/software/Kratos_MasterBranch/kratos/includes/amatrix_interface.h:24,
                 from /home/philipp/software/Kratos_MasterBranch/kratos/containers/array_1d.h:32,
                 from /home/philipp/software/Kratos_MasterBranch/kratos/geometries/point.h:30,
                 from /home/philipp/software/Kratos_MasterBranch/kratos/includes/node.h:30,
                 from /home/philipp/software/Kratos_MasterBranch/kratos/includes/element.h:27,
                 from /home/philipp/software/Kratos_MasterBranch/applications/MeshMovingApplication/custom_elements/structural_meshmoving_element.h:23,
                 from /home/philipp/software/Kratos_MasterBranch/applications/MeshMovingApplication/custom_elements/structural_meshmoving_element.cpp:18:
/home/philipp/software/Kratos_MasterBranch/external_libraries/a_matrix/include/matrix_expression.h:458:59: note: candidate: template<class TExpression1Type, class TExpression2Type, long unsigned int TCategory1, long unsigned int TCategory2> AMatrix::MatrixMinusExpression<TExpression1Type, TExpression2Type> AMatrix::operator-(const AMatrix::MatrixExpression<TExpression1Type, TCategory1>&, const AMatrix::MatrixExpression<TExpression2Type, TCategory2>&)
 MatrixMinusExpression<TExpression1Type, TExpression2Type> operator-(
                                                           ^~~~~~~~
/home/philipp/software/Kratos_MasterBranch/external_libraries/a_matrix/include/matrix_expression.h:458:59: note:   template argument deduction/substitution failed:
/home/philipp/software/Kratos_MasterBranch/applications/MeshMovingApplication/custom_elements/structural_meshmoving_element.cpp:274:75: note:   candidate expects 2 arguments, 1 provided
     noalias(rRightHandSideVector) = -prod(rLeftHandSideMatrix, last_values);
~~~

**Comment by philbucher** [comment]
@pooyan-dadvand should we merge this as is or will you implement sth in AMatrix that makes the changes unnecessary?

**Comment by philbucher** [comment]
this will work after #3514 is merged

**Comment by pooyan-dadvand** [comment]
As you like! The change was not harmful and #3514 is to avoid such problems in future.

**Comment by philbucher** [comment]
I would leave it as is, this can be merged 

Can you approve please?

**Comment by philbucher** [comment]
ping @pooyan-dadvand

**Comment by philbucher** [comment]
ping

**Review by jcotela — APPROVED** [review]

**Comment by pooyan-dadvand** [comment]
Sorry but I just missed this one....

Thanks @jcotela for approving

**Comment by philbucher** [comment]
no problem @pooyan-dadvand 
I would have waited but the changes were very minor

**Pull request merged**